src/decompression.jl

```diff
@@ -354,12 +382,28 @@ function decompress_single_color!(
 end
 
 function decompress!(A::SparseMatrixCSC, B::AbstractMatrix, result::ColumnColoringResult)
-    (; compressed_indices) = result
+    (; compressed_indices, allow_denser) = result
     S = result.bg.S2
-    check_same_pattern(A, S)
+    check_same_pattern(A, S; allow_denser)
     nzA = nonzeros(A)
-    for k in eachindex(nzA, compressed_indices)
-        nzA[k] = B[compressed_indices[k]]
+    if nnz(A) == nnz(S)
+        for k in eachindex(compressed_indices)
+            nzA[k] = B[compressed_indices[k]]
+        end
+    else  # nnz(A) > nnz(S)
+        rvA, rvS = rowvals(A), rowvals(S)
+        shift = 0
+        for j in axes(S, 2)
+            for k in nzrange(S, j)
+                i = rvS[k]
+                while (k + shift) < A.colptr[j] || rvA[k + shift] < i
```

Comment:
```suggestion
                while (k + shift) < A.colptr[j+1] || rvA[k + shift] < i
```

Reply:
@gdalle Imagine that we we are on the last column of `A` and that this column only contains nonzeros on rows < i. We will have an error because we stop for the reason `(k + shift) == A.colptr[j+1]`, which is `nnzA+1` in that case.
```
nzA[k + shift] = B[compressed_indices[k]]  # out-of-bound
```

Reply:
I don't understand why your code fixes the issue? We want to wait until we're in the right column (`k + shift >= A.colptr[j]`) and at the right row (`rvA[k + shift] = i`) before making an actual change to `A`. 
And I think the situation you imagine relies on the provided sparsity pattern being _less dense_ than `S`, because it is missing the nonzero at `(i, j)`? That is a case we explicitly don't consider